util/Migrator/DbScripts/2021-03-22_00_Send_Add_HideEmail_Column.sql

```diff
@@ -0,0 +1,154 @@
+-- Add HideEmail column
+-- Use default value of false for migration only
+ALTER TABLE [dbo].[Send]
+ADD HideEmail BIT NOT NULL CONSTRAINT hideEmail_default_value DEFAULT 0
```

Comment:
I don't see this default constraint in the Table definition... and isn't the proper way to pre-populate a NOT NULL column.

Please change this (including through line 9) to something like:

```sql
IF COL_LENGTH('[dbo].[Send]', 'HideEmail') IS NULL
BEGIN

    ALTER TABLE [dbo].[Send] ADD [HideEmail] BIT NULL;

END
GO

-- DO OTHER UPDATE STUFF HERE, like Sprocs, views, etc.

-- End of File Below
UPDATE [dbo].[Send] SET [HideEmail] = 0 WHERE [HideEmail] IS NULL;
ALTER TABLE [dbo].[Send] ALTER COLUMN [HideEmail] BIT NOT NULL;
GO
```

Keep in mind as well this kind of `NOT NULL` column change without a default value AND no default value in the SPROC/parameters AND a database deployment that occurs before the server code is fully live/published will cause creating/editing Sends to break momentarily. Not necessarily the end of the world, but something to keep in mind (I don't think we use default values anywhere else either for these `NOT NULL` columns, but personally I'm not a fan of `BIT NOT NULL` when it doesn't really matter, either it's set explicitly or it's not, the fallback to the default value of `false` is easy enough to achieve in code using `Nullable<bool>`/`bool?`.

Reply:
All good points, this has clarified my thinking on how to handle db changes. I've changed the `HideEmail` column to `NULL` and made it nullable in the models. Given that, I don't think there's any need to populate it with default values, so I've left that part out.

Reply:
A further note on this - I've made `HideEmail` nullable everywhere in the server, because there are multiple sources it could get null values from (e.g. client, database). In the clients, I've checked and fixed null values in the `SendResponseModel` only, because that's the only entry point for a null value. Let me know if I've missed something here.